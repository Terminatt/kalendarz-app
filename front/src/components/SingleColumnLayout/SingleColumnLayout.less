@import '~styles/mixins.less';
@import '~styles/variables.less';

.single-column {
    .panel-layout();
}