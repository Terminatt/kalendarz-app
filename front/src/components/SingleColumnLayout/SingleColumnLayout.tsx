import { GenericReactContent } from '@generics/generics';
import { joinClassNames } from '@utils/general';
import React from 'react';
import './SingleColumnLayout.less';

export interface SingleColumnLayoutProps {
    className?: string;
    headerText?: string;
    children: GenericReactContent
}

const SingleColumnLayout: React.FC<SingleColumnLayoutProps> = (props) => {
    const { className, headerText, children } = props;

    return (
        <div className={joinClassNames(['single-column', className])}>
            <h2>
                {headerText}
            </h2>
            {children}
        </div>
    );
};

export default SingleColumnLayout;
