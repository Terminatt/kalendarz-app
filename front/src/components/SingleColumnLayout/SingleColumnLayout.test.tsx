import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SingleColumnLayout from './SingleColumnLayout';

describe('Single Column Layout Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer
            .create(
                <SingleColumnLayout>
                    Test
                </SingleColumnLayout>
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('renders text', () => {
        const text = 'Test';
        const element = render(
            <SingleColumnLayout>
                {text}
            </SingleColumnLayout>

        );
        expect(element.getByText(text)).not.toBeNull();
    });

    it('renders header', () => {
        const text = 'Test';
        const headerText = 'Header test';
        const element = render(
            <SingleColumnLayout headerText={headerText}>
                {text}
            </SingleColumnLayout>

        );
        expect(element.getByText(headerText)).not.toBeNull();
    });
});
