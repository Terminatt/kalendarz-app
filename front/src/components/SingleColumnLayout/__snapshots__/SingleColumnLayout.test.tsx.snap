// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single Column Layout Component matches the snapshot 1`] = `
<div
  className="single-column"
>
  Test
</div>
`;
